documentation/secure_storage.md

```diff
@@ -0,0 +1,160 @@
+# Secure Storage In OP-TEE
+
+## Background
+
+Secure Storage (SST) in OP-TEE is implemented according to what has been defined
+in GloblaPlatform’s TEE Internal API specification (here called Trusted
+Storage). This specification mandates that it should be possible to store
+general-purpose data and key material that guarantees confidentiality and
+integrity of the data stored and the atomicity of the operations that modifies
+the storage (atomicity here means that either the entire operation completes
+successfully or no write is done).
+
+## Overview
+
+![Secure Storage System Architecture](images/secure_storage/secure_storage_system_architecture.png
+"Secure Storage System Architecture")
+
+### Source Files In OP-TEE OS
+
+- **core/tee/tee_svc_storage.c:** TEE trusted storage service calls
+- **core/tee/tee_fs.c:** TEE file operation interface
+- **core/tee/tee_fs_common.c:** TEE file system & REE file operation interface
+- **core/tee/tee_fs_key_manager.c:** Key manager
+- **lib/libutee/:** GlobalPlatform Internal API library
+
+### Basic File Operation Flow
+
+When TA is calling the write function provided by GP Trusted Storage API to
+write data to a persistent object, a corresponding syscall implemented in TEE
+Trusted Storage Service will be called, which in turn will invoke a series of
+TEE file operations to store the data. TEE file system will then encrypt the data
```

Comment:
will then encrypt the data

==> 

    will then encrypt the data (when CFG_ENC_FS=y)